### PR TITLE
Add WORKFLOW.md

### PR DESCRIPTION
**Pull request by RichiH**
Signed-off-by: Richard Hartmann <richih@richih.org>

### CODE REVIEW

**Review by bergquist — APPROVED** [review]
LGTM

**Comment by oddlittlebird** [comment]
I haven't finished going through the Google doc.

**Review by oddlittlebird — CHANGES_REQUESTED** [review]
Please let me finish reviewing.

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:9]:
```suggestion
Team members and their access to repositories is maintained through [GitHub teams](https://help.github.com/en/github/setting-up-and-managing-organizations-and-teams/about-teams). Team maintainers add and remove team members as outlined in GOVERNANCE.md.
```

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:12]:
```suggestion
## Code changes
```

Comment (oddlittlebird) [reply WORKFLOW.md:12]: Please remove the empty line above this paragraph.

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:16]:
```suggestion
Examples of proposed changes are overarching architecture, subsets, and specific code or graphical elements. Proposed changes SHOULD cover the big picture and intention, but individual parts SHOULD be split into the smallest possible changes. Changes SHOULD be based on and target the master branch. Depending on size of the proposed change, each change SHOULD be discussed, in increasing order of change size and complexity:
```

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:17]:
```suggestion
Directly in a pull request (PR) - This MAY be done, but SHOULD not be the common case.
```

Comment (oddlittlebird) [reply WORKFLOW.md:17]: This list SHOULD be an unordered list.

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:25]:
```suggestion

Depending on the size and complexity of a PR, different requirements MUST be applied. Any team member contributing substantially to a PR MUST NOT count against review requirements.
```

Comment (oddlittlebird) [reply WORKFLOW.md:25]: Fix the content in lines 26-33 so that they make a coherent paragraph or two. Also, end the sentences with periods.

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:35]:
```suggestion
PRs MUST be [reviewed](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/reviewing-changes-in-pull-requests) and [approved](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/approving-a-pull-request-with-required-reviews) via GitHub’s review system. 

- Reviewers MAY write comments if approving. 
- Reviewers MUST write comments if rejecting a PR or if requesting changes.
```

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:36]:
```suggestion
Once a PR is approved as per above, any team member MAY merge the PR.
```

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:39]:
```suggestion
## Release workflow
```

Comment (oddlittlebird) [reply WORKFLOW.md:39]: Delete the empty line above and adjust the heading levels that follow to be +1 depth.

File: WORKFLOW.md
Comment (oddlittlebird) [reply WORKFLOW.md:12]: Adjust the heading levels that follow this to be +1 in depth until the next h1 to h2 adjustment.

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:24]:
```suggestion
### Merging PRs
```

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:41]: Add an empty line after each heading to increase readability for maintainers.

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:44]: Formatting is unclear. It looks like the following lines should be an unordered list. If so, please add the necessary formatting and end line 44 with a colon.

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:53]:
```suggestion
Release branches MUST be split from the following branches:
```

Comment (oddlittlebird) [reply WORKFLOW.md:53]: Make the following three lines an unordered list.

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:58]: Is there a reason SECURITY.md was not linked? Serious question, not sarcastic. If not, then please link.

If there is, then ignore this comment.

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:60]:
```suggestion
PRs intended for inclusion in the next PATCH release MUST be labeled with `cherry-pick-needed` so that they can be picked up by automated release tooling.
```

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:62]: Apply unordered list formatting to the following three lines.

File: WORKFLOW.md
Comment (oddlittlebird) [inline WORKFLOW.md:69]: In this context, what is a "relevant subsystem maintainer"? Consider listing a few examples.

**Review by oddlittlebird — CHANGES_REQUESTED** [review]

File: WORKFLOW.md
Comment (RichiH) [reply WORKFLOW.md:69]: I can refer to https://github.com/grafana/grafana/blob/master/MAINTAINERS.md , though that's already done in governance, and we refer to governance in its entirety.

File: WORKFLOW.md
Comment (RichiH) [reply WORKFLOW.md:58]: Getting SECURITY.md in place in WIP, but I didn't want to block on that.

**Review by RichiH — COMMENTED** [review]
Thanks @oddlittlebird - the unordered lists got lost in copy & paste and I missed it. Adapted newlines around heading as per your preference. Will force-push a squash of your and my changes.

@bergquist argued we should avoid abbrevations like PR, I like them, both is fine by me.

**Comment by oddlittlebird** [comment]
> @bergquist argued we should avoid abbrevations like PR, I like them, both is fine by me.

We define the abbreviation on first usage and "PR" is accepted GitHub terminology, albeit not in the glossary (an oversight on GitHub's part, imo). It's kosher, but I do not feel super strongly either way.

**Review by oddlittlebird — CHANGES_REQUESTED** [review]
Please finish applying or resolving proposed changes. There should only be one h1 per topic.

**Review by daniellee — APPROVED** [review]
LGTM. Just wondered about the subset reference in the Proposing changes section.

File: WORKFLOW.md
Comment (bergquist) [reply WORKFLOW.md:55]: That's why it says `MUST` and `SHOULD`

File: WORKFLOW.md
Comment (RichiH) [reply WORKFLOW.md:15]: Fixed in new commit.

File: WORKFLOW.md
Comment (RichiH) [reply WORKFLOW.md]: Adapted.

File: WORKFLOW.md
Comment (RichiH) [reply WORKFLOW.md:11]: Newline removed, heading is how I think the structure makes sense.

**Review by oddlittlebird — APPROVED** [review]
LGTM

**Pull request merged**
